testing.rb

```diff
@@ -0,0 +1,313 @@
+          return false if n.match?(input)
+        end
+      end
+      true
+    elsif !block_given? && input.nil?
+      my_each do |item|
+        return false if item == true
+      end
+      true
+    elsif block_given?
+      my_each do |item|
+        return false if item == true
+        return false if yield(item)
+      end
+      true
+    end
+  end
+
+  def my_count(*args)
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+
+    if !block_given? && args[0].nil?
+      counter = 0
+      counter += 1 while counter < arr.length
+      counter
+    elsif !block_given? && !args[0].nil?
+      count = 0
+      section = 0
+      while count < length
+        section += 1 if arr[count] == args[0]
+        count += 1
+      end
+      section
+    elsif block_given? && args[0].nil?
+      num = 0
+      ematch = 0
+      while num < length
+        ematch += 1 if yield(arr[num])
+        num += 1
+      end
+      ematch
+    end
+  end
+
+  def my_map(args = nil)
+    return to_enum(:my_map) unless block_given?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+
+    if block_given? && !args.nil?
+      results = []
+      my_each { |x| results.push(args[x]) }
+      results
+    elsif block_given? && args.nil?
+      counter = 0
+      tot = []
+      while counter < arr.length
+        runn = yield(arr[counter])
+        tot.push(runn)
+        counter += 1
+      end
+      tot
+    end
+  end
+
+  def my_inject(args = nil, arg = nil)
+    return to_enum(:my_inject) if !block_given? && args.nil? && arg.nil?
+
+    arr = to_a if self.class == Range
+    arr = self if self.class == Array
+    if !block_given? && !args.nil? && arg.nil? # no blocks and args
+      i = 1
+      x = 0
+      acum = arr[x]
+      while i < arr.length
+        acum = acum.send(args, arr[i])
+        i += 1
+      end
+      acum
+    elsif block_given? && args.nil? && args.nil?
+      if arr[0].class == String
+        counter = 0
+        longest = nil
+        while counter < arr.length
+          longest = yield(arr[counter], arr[counter + 1]) unless arr[counter + 1].nil?
+          counter += 1
+        end
+        longest
+      elsif arr[0].class == Integer
+        sum = 1
+        num = 0
+        total = arr[num]
+        while sum < arr.length
+          total = yield(total, arr[sum])
+          sum += 1
+          num += 1
+        end
+        total
+      end
+    elsif block_given? && !args.nil? && arg.nil? # blocks and args
+      sum = 1
+      num = 0
+      totals = arr[num]
+      while sum < arr.length
+        totals = yield(totals, arr[sum])
+        sum += 1
+        num += 1
+      end
+      yield(totals, args)
+    elsif !block_given? && !args.nil? && !arg.nil? # no blocks and args
+      i = 1
+      x = 0
+      acum = arr[x]
+      while i < arr.length
+        acum = acum.send(arg, arr[i])
+        i += 1
+      end
+      acum.send(arg, args)
+    end
+  end
```

Comment:
Please make sure your `my_inject` returns a proper error message. Here's the error from our automated tests:

```
5) enumerables #my_inject raises a "LocalJumpError" when no block or argument is given
     Failure/Error: expect { array.my_inject }.to raise_error(LocalJumpError)
       expected LocalJumpError but nothing was raised
```

Find out how to raise a `LocalJumpError` in ruby and implemented as requested in the error message.